testsuite/tests/typing-modal-kinds/everything.ml

```diff
@@ -0,0 +1,291 @@
+[%%expect{|
+Line 1, characters 0-44:
+1 | type bad : float64 mod contended = t_float64
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_float64" is float64
+         because of the definition of t_float64 at line 3, characters 0-24.
+       But the kind of type "t_float64" must be a subkind of
+         float64 mod contended
+         because of the definition of bad at line 1, characters 0-44.
+|}]
+
+type good : float64 mod contended = t_float64_mod_e
+[%%expect{|
+type good = t_float64_mod_e
+|}]
+
+(*** yielding ***)
+type bad : value mod unyielding = t_value
+[%%expect{|
+Line 1, characters 0-41:
+1 | type bad : value mod unyielding = t_value
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_value" is value
+         because of the definition of t_value at line 1, characters 0-20.
+       But the kind of type "t_value" must be a subkind of value mod unyielding
+         because of the definition of bad at line 1, characters 0-41.
+|}]
+
+type good : value mod unyielding = t_value_mod_e
+[%%expect{|
+type good = t_value_mod_e
+|}]
+
+type bad : float64 mod unyielding = t_float64
+[%%expect{|
+Line 1, characters 0-45:
+1 | type bad : float64 mod unyielding = t_float64
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_float64" is float64
+         because of the definition of t_float64 at line 3, characters 0-24.
+       But the kind of type "t_float64" must be a subkind of
+         float64 mod unyielding
+         because of the definition of bad at line 1, characters 0-45.
+|}]
+
+type good : float64 mod unyielding = t_float64_mod_e
+[%%expect{|
+type good = t_float64_mod_e
+|}]
+
+(*** external ***)
+type bad : value mod external_ = t_value
+[%%expect{|
+Line 1, characters 0-40:
+1 | type bad : value mod external_ = t_value
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_value" is value
+         because of the definition of t_value at line 1, characters 0-20.
+       But the kind of type "t_value" must be a subkind of value mod external_
+         because of the definition of bad at line 1, characters 0-40.
+|}]
+
+type good : value mod external_ = t_value_mod_e
+[%%expect{|
+type good = t_value_mod_e
+|}]
+
+type bad : float64 mod external_ = t_float64
+[%%expect{|
+Line 1, characters 0-44:
+1 | type bad : float64 mod external_ = t_float64
+    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Error: The kind of type "t_float64" is float64
+         because of the definition of t_float64 at line 3, characters 0-24.
+       But the kind of type "t_float64" must be a subkind of
+         float64 mod external_
+         because of the definition of bad at line 1, characters 0-44.
+|}]
+
+type good : float64 mod external_ = t_float64_mod_e
+[%%expect{|
+type good = t_float64_mod_e
+|}]
+
+(* mod everything does not affect nullability *)
+type t : value_or_null mod everything
+type bad : value = t
```

Comment:
how does this test that `mod everything` doesn't affect nullability?

Reply:
I think if `mod everything` set the bound of `t` to `non_null` then the second line would typecheck.

Reply:
I agree with Chris